quinn-proto/src/connection/mod.rs

```diff
@@ -710,12 +700,23 @@ where
                     self.handle_coalesced(now, remote, ecn, data);
                 }
             }
-            NewIdentifiers(ids) => {
-                self.cids_issued += ids.len() as u64;
-                ids.into_iter().rev().for_each(|frame| {
-                    self.spaces[SpaceId::Data].pending.new_cids.push(frame);
-                    self.cids_active_seq.insert(frame.sequence);
-                });
+            NewIdentifiers(ids, now) => {
+                // `ids` could be `None` once active_connection_id_limit is set to 1 by peer
+                if let Some(last_cid) = ids.last() {
+                    self.local_cid_state.issued += ids.len() as u64;
+                    // Record the timestamp of CID with the largest seq number
+                    let sequence = last_cid.sequence;
+                    ids.into_iter().rev().for_each(|frame| {
+                        self.spaces[SpaceId::Data].pending.new_cids.push(frame);
+                        self.local_cid_state.active_seq.insert(frame.sequence);
+                    });
+                    self.local_cid_state.track_lifetime(sequence, now);
+                }
```

Comment:
This is manipulating an awful lot of `self.local_cid_state` stuff. Should we move more of this code into a `CidState` method?

Reply:
Ack